### PR TITLE
[AEROGEAR-2592] Show latest build and build history

### PR DESCRIPTION
**Pull request by JameelB**
## Description
Show the latest build and build histories with a link and a qr code for the artefact download.

## Progress
- [x] Create a section for the latest build within the mobile client build row
- [x] Create a section for the build history within the mobile client build row
- [x] Display artefact download link and qr code for the latest build and build history

## Verification steps
**Setup**
- Provision a mobile client
- Verify that the `Builds` tab is not available within the mobile client view as there is no mobile ci/cd provisioned
- Provision a mobile ci/cd service
- Verify that the `Builds` tab is available within the mobile client view.
- Verify that a message shows within the `Builds` tab stating `No Build Config`

**Creating Builds**
- Create a build
- Verify that a new build is created and shown within the build tab
- Verify that there is a `Start Build` button available in the header
- Expand the build to see the build overview
- Verify that a `Builds` section is available
- Verify that a message stating `No Builds` is shown and a `Start Build` button is displayed within the `Builds` section.
- Start the build
- Verify that a success notification appears
- Verify that a build process has started and the latest build is displayed within the `Builds` section with the pipeline stages.
- Verify that the status icon within the build view and the header corresponds correctly to the build process status
- Verify that the links direct to the correct places.
- Verify that a download button appears only when the build completes successfuly
- Click the download button
- Verify that the URL is correct and a qr code is displayed

**Build History**
- Verify that `Show build history` is available
- Expand the build history
- Verify that the previous builds are shown and a download button is only available for builds that completed successfully
- Verify that when the `Download` button is clicked, the URL and qr code for the artefact is shown for that build

**Cleanup**
- Verify that a build can be deleted within the `Builds` tab

### CODE REVIEW

**Comment by odra** [comment]
:eyes:

**Comment by odra** [comment]
@JameelB the download link is returning the wrong http protocol (should be using https instead of http in my case) - is this something in the ui code or are you just using whats the artifact proxy returns from its api?

File: app/views/mobile-client-download-url.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div ng-if="$ctrl.showPanel" class="mobile-client-download-url col-md-12">
+  <div ng-if="!$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]" class="spinner spinner-lg" aria-hidden="true"></div>
```
Comment (dimitraz) [inline app/views/mobile-client-download-url.html]: There's a filter for getting an annotation that can be used in the html which might shorten this a bit. 
Alternatively, what about extracting this to a variable like `isAnnotated` in the controller, seeing as its being used quite a bit?

File: app/views/mobile-client-download-url.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <div ng-if="$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]" class="col-md-6 build-url">
+    <p>Download from URL:</p>
+    <a ng-href="{{$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]}}">{{$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]}}</a>
+  </div>
+  <div ng-if="$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]" class="col-md-6 build-qrcode">
```
Comment (dimitraz) [inline app/views/mobile-client-download-url.html]: Could the two `ng-ifs` here and above be extracted so a single container div?

**Comment by dimitraz** [comment]
Is the download link supposed to be working? It doesn't seem to work for me but maybe I'm doing something wrong. I'm getting this error from the artifact proxy:
`no build type found for build &Build{....}` 

@philbrookes  @philipgough

**Comment by odra** [comment]
@JameelB nevermind I had an outdated version of the artifact proxy image but I got the same error as @dimitraz

**Review by maleck13 — APPROVED** [review]

**Comment by odra** [comment]
@JameelB The BuildConfig object needs the following label item:

`mobile-client-type: $platform` - $platform being "android" or "ios"

File: app/views/mobile-client-download-url.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    <span> Or scan QR code: </span>
+    <jq-qr-code content="$ctrl.mobileBuild.metadata.annotations[$ctrl.BUILD_DOWNLOAD_ANNOTATION]"/>
```
Comment (dimitraz) [inline app/views/mobile-client-download-url.html]: Is it possible to make the qr code a bit smaller to match the invisionapp?

**Review by dimitraz — APPROVED** [review]
Verified locally :+1:

**Review by odra — APPROVED** [review]

**Review by vchepeli — APPROVED** [review]

**Pull request merged**